test/composableStableJoin.integration.test.ts

```diff
@@ -0,0 +1,283 @@
+            );
+            const poolTokensWithoutBpt = txInput.poolStateInput.tokens
+                .map((t) => new Token(chainId, t.address, t.decimals))
+                .filter((_, index) => index !== bptIndex);
+
+            amountsIn = poolTokensWithoutBpt.map((t) =>
+                TokenAmount.fromHumanAmount(t, '1'),
+            );
+            input = {
+                chainId,
+                rpcUrl,
+                kind: JoinKind.Unbalanced,
+            };
+        });
+        test('token inputs', async () => {
+            const joinInput = {
+                ...input,
+                amountsIn: [...amountsIn.splice(0, 1)],
+            };
+            const joinResult = await doJoin({
+                ...txInput,
+                joinInput,
+            });
+            assertUnbalancedJoin(
+                txInput.client.chain?.id as number,
+                txInput.poolStateInput,
+                joinInput,
+                joinResult,
+                txInput.slippage,
+            );
+        });
+
+        test('with native', async () => {
+            const joinInput = {
+                ...input,
+                amountsIn,
+                useNativeAssetAsWrappedAmountIn: true,
+            };
+            const joinResult = await doJoin({
+                ...txInput,
+                joinInput: {
+                    ...joinInput,
+                    useNativeAssetAsWrappedAmountIn: true,
+                },
+            });
+            assertUnbalancedJoin(
+                txInput.client.chain?.id as number,
+                txInput.poolStateInput,
+                { ...joinInput, useNativeAssetAsWrappedAmountIn: true },
+                joinResult,
+                txInput.slippage,
+            );
+        });
+    });
+
+    describe('single asset join', () => {
+        let joinInput: SingleAssetJoinInput;
+        beforeAll(() => {
+            const bptOut = TokenAmount.fromHumanAmount(bptToken, '1');
+            const tokenIn = '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2';
+            joinInput = {
+                bptOut,
+                tokenIn,
+                chainId,
+                rpcUrl,
+                kind: JoinKind.SingleAsset,
+            };
+        });
+        test('with token', async () => {
+            const joinResult = await doJoin({
+                ...txInput,
+                joinInput,
+            });
+
+            assertSingleToken(
+                txInput.client.chain?.id as number,
+                txInput.poolStateInput,
+                joinInput,
+                joinResult,
+                txInput.slippage,
+            );
+        });
+
+        test('with native', async () => {
+            const joinResult = await doJoin({
+                ...txInput,
+                joinInput: {
+                    ...joinInput,
+                    useNativeAssetAsWrappedAmountIn: true,
+                },
```

Comment:
If you redeclare `joinInput` with this flag before `joinResult`, you should be able to reuse as you did on the previous test.